ios/FluentUI/Table View/TableViewCell.swift

```diff
@@ -404,18 +434,42 @@ open class TableViewCell: UITableViewCell, TokenizedControlInternal {
         let trailingAccessoryAreaWidth = Self.labelTrailingAccessoryAreaWidth(viewWidth: trailingAccessoryViewWidth,
                                                                               text: text, labelAccessoryViewMarginLeading: labelAccessoryViewMarginLeading)
 
-        let availableWidth = textAreaWidth - (leadingAccessoryAreaWidth + trailingAccessoryAreaWidth)
-
+        let availableWidth = textAreaWidth - (leadingAccessoryAreaWidth + trailingAccessoryAreaWidth + labelAccessoryViewMarginTrailing)
+        if isAttributedTextSet, let attributedText = attributedText {
+            return preferredLabelSize(with: attributedText, availableTextWidth: availableWidth)
+        }
         return text.preferredSize(for: font, width: availableWidth, numberOfLines: numberOfLines)
     }
 
+    private static func preferredLabelSize(with attributedText: NSAttributedString,
+                                           availableTextWidth: CGFloat = .greatestFiniteMagnitude) -> CGSize {
+        let estimatedBoundsWidth = attributedText.boundingRect(
```

Comment:
feel like this needs lots of comments why?

Reply:
Adding comments. This is needed because the .usesDeviceMetrics option will cut off the bottom portion of the label's text, but removing it will cut off the trailing edge of the text. Making two rect values was my solution